src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Extensions/ViewCssScopePass.cs

```diff
@@ -56,29 +56,28 @@ private void ProcessElement(HtmlContentIntermediateNode node, string cssScope)
             }
         }
 
-        bool IsValidElement(IntermediateToken token)
+        static bool IsValidElement(IntermediateToken token)
         {
             var content = token.Content;
-            var isValidToken = content.StartsWith("<", StringComparison.Ordinal)
-                && !content.StartsWith("</", StringComparison.Ordinal)
-                && !content.StartsWith("<!", StringComparison.Ordinal);
             /// <remarks>
             /// We want to avoid adding the CSS scope to elements that do not appear
             /// within the body element of the document. When this pass executes over the
-            /// nodes, we don't have the ability to store whether we are a descandant of a
+            /// nodes, we don't have the ability to store whether we are a descendant of a
             /// `head` or `body` element so it is not possible to discern whether the tag
             /// is valid this way. Instead, we go for a straight-forward check on the tag
             /// name that we are currently inspecting.
             /// </remarks>
```

Comment:
Note that we seem to be doing this only for `.cshtml` views, not `.razor` components which seems inconsistent but that's an unrelated issue.